src/lsp/client.ts

```diff
@@ -1,29 +1,39 @@
 import type Monaco from "monaco-editor-core";
 import type { Workspace } from "~/workspace.ts";
 import * as lst from "vscode-languageserver-types";
+import type { WorkspaceInit } from "../../types/workspace";
 
 // ! external modules, don't remove the `.js` extension
 import { cache } from "../cache.js";
 
-let monaco: typeof Monaco;
+const monacoAPI = {
+  get monaco() {
+    return (globalThis as any).MonacoEnvironment?.monaco as typeof Monaco;
+  }
+};
 
-/** init the monaco namespace. */
-export function init(monacoNS: typeof Monaco): void {
```

Comment:
No need for this init function anymore, as we now set the monaco instance on the MonacoEnvironment (in core.ts)